pkg/command/context.go

```diff
@@ -1106,95 +1130,94 @@ func displayContextListOutputListView(cfg *configtypes.ClientConfig, writer io.W
 		}
 		contextOutputList = append(contextOutputList, ctx)
 	}
 	return contextOutputList, hasTAEFields
 }
 
-func displayContextListOutputSplitViewTarget(cfg *configtypes.ClientConfig, writer io.Writer) {
-	var k8sContextTable component.OutputWriter
-	target := getTarget()
+type ContextListOutputRow struct {
+	Name           string
+	IsActive       string
+	Type           string
+	Endpoint       string
+	KubeconfigPath string
+	KubeContext    string
+	Project        string
+	Space          string
+}
 
-	ctxs, showTAEColumns := getContextsToDisplay(cfg, target, onlyCurrent)
+func displayContextListOutputWithDynamicColumns(cfg *configtypes.ClientConfig, writer io.Writer) {
+	ct := getContextType()
+	ctxs, _ := getContextsToDisplay(cfg, ct, onlyCurrent)
 
 	opts := []component.OutputWriterOption{}
-	if showTAEColumns {
-		k8sContextTable = component.NewOutputWriterWithOptions(writer, outputFormat, opts, "Name", "IsActive", "Type", "Endpoint", "KubeConfigPath", "KubeContext", "Project", "Space")
-	} else {
-		k8sContextTable = component.NewOutputWriterWithOptions(writer, outputFormat, opts, "Name", "IsActive", "Type", "Endpoint", "KubeConfigPath", "KubeContext")
-	}
-	outputWriterTMCTarget := component.NewOutputWriterWithOptions(writer, outputFormat, opts, "Name", "IsActive", "Endpoint")
+	rows := []ContextListOutputRow{}
 	for _, ctx := range ctxs {
-		var ep, path, context string
-		var project, space string
+		ep := NA
+		path := NA
+		context := NA
+		project := NA
+		space := NA
 
-		isCurrent := ctx.Name == cfg.CurrentContext[ctx.Target]
+		isCurrent := ctx.Name == cfg.CurrentContext[ctx.ContextType]
 
-		switch ctx.Target {
-		case configtypes.TargetTMC:
+		switch ctx.ContextType {
+		case configtypes.ContextTypeTMC:
 			if ctx.GlobalOpts != nil {
 				ep = ctx.GlobalOpts.Endpoint
 			}
-
-			outputWriterTMCTarget.AddRow(ctx.Name, isCurrent, ep)
-		default:
+		case configtypes.ContextTypeTAE:
+			project = ""
+			space = ""
+			ep = ""
+			path = ""
+			context = ""
 			if ctx.ClusterOpts != nil {
 				ep = ctx.ClusterOpts.Endpoint
 				path = ctx.ClusterOpts.Path
 				context = ctx.ClusterOpts.Context
 			}
-			contextType := "cluster"
-			if ctx.Target == configtypes.TargetTAE {
-				contextType = "TAE"
+			if ctx.AdditionalMetadata[taert.ProjectNameKey] != nil {
+				project = ctx.AdditionalMetadata[taert.ProjectNameKey].(string)
 			}
-
-			if showTAEColumns {
-				if ctx.AdditionalMetadata["taeProjectName"] != nil {
-					project = ctx.AdditionalMetadata["taeProjectName"].(string)
-				}
-				if ctx.AdditionalMetadata["taeSpaceName"] != nil {
-					space = ctx.AdditionalMetadata["taeSpaceName"].(string)
-				}
-				k8sContextTable.AddRow(ctx.Name, isCurrent, contextType, ep, path, context, project, space)
-			} else {
-				k8sContextTable.AddRow(ctx.Name, isCurrent, contextType, ep, path, context)
+			if ctx.AdditionalMetadata[taert.SpaceNameKey] != nil {
+				space = ctx.AdditionalMetadata["taeSpaceName"].(string)
```

Comment:
Thanks for replacing the hard-code "taeProjectName" strings with constants.  There is one more just above.

Reply:
Fixed.